src/EppoClient.php

```diff
@@ -361,16 +377,183 @@ private function getAssignmentDetail(
         return $computedVariation;
     }
 
-    private function checkExpectedType(VariationType $expectedVariationType, $typedValue): bool
-    {
+    /**
+     * Selects a Bandit action, if applicable, based on the flag key, subject, and actions provided.
+     *
+     * Actions are selected based on the subject and action contexts. Contexts are passed as associative arrays
+     * (key=>value) of attributes which, by default, are sorted into numeric and non-numeric. Non-numeric attributes
+     * are bucketed as "Categorical Attributes" while numeric are treated as "Numeric Attributes". Demonstrated in the
+     * **Example 2** below is a means of explicitly classifying attributes as either Numeric or Categorical.
+     *
+     * Logging selected bandits to your data warehouse is critical. Instead of just implementing `LoggerInterface`,
+     * implement `IBanditLogger`. This interface logging methods for both flag assignments and bandits.
+     *
+     * Example 1:
+     *
+     * $flagKey = 'my-bandit-flag';
+     * $subject = 'user-123';
+     * $subjectContext = ['accountAge' => 0.5, 'country' => 'US'];
+     *
+     * // A simple list of actions with no context attributes
+     * $actions = ['nike', 'adidas', 'reebok'];
+     *
+     * $result = $client->getBanditAction($flagKey, $subject, $subjectContext, $actions, 'control');
+     *
+     * Example 1.2: Actions with un-grouped attributes
+     * $actions = [
+     *   'nike': [
+     *     'brandLoyalty' => 0.0,
+     *     'size' => 5,
+     *     'colour' => 'red'
+     *   ], ...
+     * ];
+     *
+     * $result = $client->getBanditAction($flagKey, $subject, $subjectContext, $actions, 'control');
+     *
+     *
+     * Example 2:
+     *
+     * $subjectContext = new AttributeSet(
+     *      numericAttributes: ['accountAge' => 0.5],
+     *      categoricalAttributes: ['zip' => 90210, 'country' => 'US']
+     * );
+     *
+     * $actions = [
+     *   'nike': new AttributeSet(
+     *     numericAttributes: ['brandLoyalty' => 0.0],
+     *     categoricalAttributes: ['size' => 5, 'colour' => 'red']
+     *   ), ...
+     * ];
+     *
+     * $result = $client->getBanditAction($flagKey, $subject, $subjectContext, $actions, 'control');
+     *
+     * @param string $flagKey
+     * @param string $subjectKey
+     * @param array<string, ?object>|AttributeSet $subjectContext
+     * @param array<string>|array<string, array<string, ?object>>|array<string, AttributeSet> $actions
+     * @param string $defaultVariation
+     * @return BanditResult
+     *
+     * @throws EppoClientException
+     */
+    public function getBanditAction(
+        string $flagKey,
+        string $subjectKey,
+        array|AttributeSet $subjectContext,
+        array $actions,
+        string $defaultVariation
```

Comment:
we call this `$defaultValue` in other SDKs, thinking that may be a more appropriate name, as it may not necessarily be a variation 🤷‍♂️